### ISSUE TITLE
"Specialized molecule splitting" warning raised in some cases on going from AMBER to GROMACS format

### ISSUE THREAD

**Issue by davidlmobley** [ParmEd/ParmEd#207]
I'm taking AMBER format parameter and coordinate files for a mixture of toluene, benzene, cyclohexane and ethane (it just happens to be a test case we were running) and trying to write to GROMACS format, but I get an exception about specialized molecule splitting not being implemented yet. I haven't done any splitting, so this seems to be a bug. Input files are linked below.

```
In [20]: structure = parmed.amber.AmberParm( 'toluene_benzene_cyclohexane_ethane_3_1_80_7.prmtop', 'toluene_benzene_cyclohexane_ethane_3_1_80_7.inpcrd' )

In [21]: gromacs_topology = parmed.gromacs.GromacsTopologyFile.from_structure( structure )

In [22]: gromacs_topology.write( 'tmp.top', 'tmp.gro')
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-22-8bd3351aad6b> in <module>()
----> 1 gromacs_topology.write( 'tmp.top', 'tmp.gro')

/Users/dmobley/anaconda/lib/python2.7/site-packages/parmed/gromacs/gromacstop.pyc in write(self, dest, combine, parameters)
   1335                                     params, parameters == 'inline')
   1336             else:
-> 1337                 raise NotImplementedError('Specialized molecule splitting is '
   1338                                           'not yet supported')
   1339         finally:

NotImplementedError: Specialized molecule splitting is not yet supported
```

Input files to reproduce this: https://dl.dropboxusercontent.com/u/3409095/splitting_problem.tar.gz

**Comment by swails** [comment]
I will look into this when I get a chance. I'm traveling most of the day. But I'll try to find some time to fix it.

**Comment by swails** [comment]
So this is actually an improvement I should make in the API.  The call signature of `GromacsTopologyFile.write` doesn't take a GRO file name:

```Python
>>> help(parmed.gromacs.GromacsTopologyFile.write)
write(self, dest, combine=None, parameters='inline') unbound parmed.gromacs.gromacstop.GromacsTopologyFile method
    Write a Gromacs Topology File from a Structure
    
    Parameters
    ----------
    dest : str or file-like
        The name of a file or a file object to write the Gromacs topology to
    combine : 'all', None, or list of iterables, optional
        If None, no molecules are combined into a single moleculetype. If
        'all', all molecules are combined into a single moleculetype.
        Otherwise, the list of molecule indices will control which atoms are
        combined into single moleculetype's. Default is None
    parameters : 'inline' or str or file-like object, optional
        This specifies where parameters should be printed. If 'inline'
        (default), the parameters are written on the same lines as the
        valence terms are defined on. Any other string is interpreted as a
        filename for an ITP that will be written to and then included at the
        top of `dest`. If it is a file-like object, parameters will be
        written there.  If parameters is the same as ``dest``, then the
        parameter types will be written to the same topologyfile.
    
    Raises
    ------
    ValueError if the same molecule number appears in multiple combine lists
```

So you passed a GRO file name as the `combine` argument.  The fix is to write these separately:

```Python
gromacs_topology.write('tmp.top')
parmed.gromacs.GromacsGroFile.write(gromacs_topology, 'tmp.gro')
```

I want to allow `GromacsTopologyFile` to both read and write a GRO file (in much the same way that `AmberParm` works).  But for the time being, ParmEd is behaving as designed here.

I went ahead and added the enhancement label as a reminder to myself to come back to this.

**Comment by davidlmobley** [comment]
Oops. Sorry about that. And I'd done this correctly elsewhere, but
apparently I forgot to look at my own code. Thanks for looking at it.

On Thu, Jun 18, 2015 at 4:39 PM, Jason Swails <notifications@github.com>
wrote:

> So this is actually an improvement I should make in the API. The call
> signature of GromacsTopologyFile.write doesn't take a GRO file name:
>
> >>> help(parmed.gromacs.GromacsTopologyFile.write)
> write(self, dest, combine=None, parameters='inline') unbound parmed.gromacs.gromacstop.GromacsTopologyFile method
>     Write a Gromacs Topology File from a Structure
>
>     Parameters
>     ----------
>     dest : str or file-like
>         The name of a file or a file object to write the Gromacs topology to
>     combine : 'all', None, or list of iterables, optional
>         If None, no molecules are combined into a single moleculetype. If
>         'all', all molecules are combined into a single moleculetype.
>         Otherwise, the list of molecule indices will control which atoms are
>         combined into single moleculetype's. Default is None
>     parameters : 'inline' or str or file-like object, optional
>         This specifies where parameters should be printed. If 'inline'
>         (default), the parameters are written on the same lines as the
>         valence terms are defined on. Any other string is interpreted as a
>         filename for an ITP that will be written to and then included at the
>         top of `dest`. If it is a file-like object, parameters will be
>         written there.  If parameters is the same as ``dest``, then the
>         parameter types will be written to the same topologyfile.
>
>     Raises
>     ------
>     ValueError if the same molecule number appears in multiple combine lists
>
> So you passed a GRO file name as the combine argument. The fix is to
> write these separately:
>
> gromacs_topology.write('tmp.top')
> parmed.gromacs.GromacsGroFile.write(gromacs_topology, 'tmp.gro')
>
> I want to allow GromacsTopologyFile to both read and write a GRO file (in
> much the same way that AmberParm works). But for the time being, ParmEd
> is behaving as designed here.
>
> I went ahead and added the enhancement label as a reminder to myself to
> come back to this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ParmEd/ParmEd/issues/207#issuecomment-113319525>.
>



-- 
David Mobley
dmobley@gmail.com
949-385-2436

**Comment by swails** [comment]
No worries.  Your usage was an intuitive one, and I plan on adjusting the API to behave that way very soon.

**Comment by swails** [comment]
This was implemented in #218.
